roles/openshift_metrics/tasks/install_hawkular.yaml

```diff
@@ -47,18 +47,34 @@
   slurp: src={{ openshift_metrics_certs_dir }}/ca.crt
   register: metrics_route_dest_ca_cert
 
-- name: generate the hawkular-metrics route
-  template:
-    src: route.j2
-    dest: "{{ mktemp.stdout }}/templates/hawkular-metrics-route.yaml"
-  vars:
-    name: hawkular-metrics
-    labels:
-      metrics-infra: hawkular-metrics
-    host: "{{ openshift_metrics_hawkular_metrics_hostname }}"
-    to:
-      kind: Service
+- block:
+  - local_action: shell mktemp
+    register: ca_tempfile
+    changed_when: False
+
+  - local_action: stat path={{openshift_metrics_hawkular_ca}}
+    register: hawkular_ca_file
+    changed_when: False
+
+  - local_action: shell cat {{openshift_metrics_hawkular_ca}} > {{ca_tempfile.stdout}}
+    when: hawkular_ca_file.stat.exists 
+
+  - name: generate the hawkular-metrics route
+    template:
+      src: route.j2
+      dest: "{{ mktemp.stdout }}/templates/hawkular-metrics-route.yaml"
+    vars:
       name: hawkular-metrics
-    tls:
-      termination: reencrypt
-      destination_ca_certificate: "{{ metrics_route_dest_ca_cert.content }}"
+      labels:
+        metrics-infra: hawkular-metrics
+      host: "{{ openshift_metrics_hawkular_hostname }}"
+      to:
+        kind: Service
+        name: hawkular-metrics
+      tls:
+        termination: reencrypt
+        ca_certificate: "{{ lookup('file',ca_tempfile.stdout) }}"
+        destination_ca_certificate: "{{ metrics_route_dest_ca_cert.content | b64decode }}"
+  always:
+  - local_action: file state=absent path={{ca_tempfile.stdout}}
+    changed_when: False
```

Comment:
The temp file was introduced to handle the the hawkular CA being provided by the user..  The change with these tasks allow someone deploying metrics to, theoretically, run from a control node and not have to be on the master.  If you try to load the CA and none is provided, the tasks will fail.  This sequence allows you to either provide nothing and process an empty file, or provide the file.  It ends up being a hack IMO to handle a case that ansible doesnt seem to handle well.